### DISCUSSION TITLE
webGPU: only for actual GPUs or also available for integrated graphics in CPUs?

### DISCUSSION THREAD

**Discussion by rocco-haro** [gpuweb/gpuweb#4098] [Q&A] [answered]
I've been following the developments and i'm excited about the opportunities webGPU will unlock. After combing through some forums and the proposed technical drafts I couldn't find any language that touched on integrated graphics. I'm no expert in hardware systems, but i've come to learn that there are CPUs with GPUs integrated directly into their system - integrated graphics. 

Obviously integrated graphics is no where near as powerful as a consumer gaming GPU, but I believe that there could still be significant web app performance gains if we could still access integrated graphics by pushing work to it instead of just letting the browser decide how to perform certain computations. 

This would be a way for users who don't have a consumer GPU to still get some level of improvement relative to the traditional way. 

Well, that was my long-winded way of asking: will webGPU offer access to integrated graphics too?

**Comment by Kangz** [comment, marked as answer]
TL;DR: integrated GPUs are supported.

Integrated graphics have been discussed extensively and are definitely supported by WebGPU. For example [`GPUPowerPreference`](https://gpuweb.github.io/gpuweb/#enumdef-gpupowerpreference) is made for the Web page to hint which GPU of a dual GPU it needs, and the buffer mapping design is more complex because it takes into account integrated GPUs.

**Reply by kainino0x** [reply]
@rocco-haro In typical parlance, a GPU is a GPU regardless of whether it's integrated or discrete. In fact, the vast majority of devices out there _only_ have integrated GPUs: phones, tablets, and most laptops. Discrete GPUs (on separate "video cards") are mostly only found in desktops and higher-end/larger/higher-power laptops.

**Reply by rocco-haro** [reply]
Thank you @Kangz for the answer, and @kainino0x for the clarification!
